packages/typescript-estree/src/convert.ts

```diff
@@ -3319,23 +3319,33 @@ export class Converter {
     aliasKey: AliasKey,
     valueKey: ValueKey,
   ): Properties & Record<AliasKey, Properties[ValueKey]> {
-    let errored = false;
+    if (this.options.suppressDeprecatedPropertyWarnings) {
+      (node as any)[aliasKey] = node[valueKey];
+      return node as Properties & Record<AliasKey, Properties[ValueKey]>;
+    }
+
+    let warned = false;
 
     Object.defineProperty(node, aliasKey, {
-      get: this.options.suppressDeprecatedPropertyWarnings
-        ? (): Properties[typeof valueKey] => node[valueKey]
-        : (): Properties[typeof valueKey] => {
-            if (!this.options.suppressDeprecatedPropertyWarnings) {
-              if (!errored) {
-                // eslint-disable-next-line no-console
-                console.warn(
-                  `The '${aliasKey}' property is deprecated on ${node.type} nodes. Use '${valueKey}' instead. See https://typescript-eslint.io/linting/troubleshooting#the-key-property-is-deprecated-on-type-nodes-use-key-instead-warnings.`,
-                );
-                errored = true;
-              }
-            }
-            return node[valueKey];
-          },
+      configurable: true,
+      get(): Properties[typeof valueKey] {
+        if (!warned) {
+          // eslint-disable-next-line no-console
+          console.warn(
+            `The '${aliasKey}' property is deprecated on ${node.type} nodes. Use '${valueKey}' instead. See https://typescript-eslint.io/linting/troubleshooting#the-key-property-is-deprecated-on-type-nodes-use-key-instead-warnings.`,
+          );
+          warned = true;
+        }
+
+        return node[valueKey];
+      },
+      set(value): void {
+        Object.defineProperty(node, aliasKey, {
+          enumerable: true,
+          writable: true,
+          value,
+        });
+      },
```

Comment:
@fisker - does this set of changes work for you?

Reply:
This should work for us.

Reply:
Ah, hmm, setting `node[aliasKey] = node[valueKey]` does again make the node properties show up when enumerating - including in AST viewers and our snapshots. @fisker would it work to still always `Object.defineProperty`, with the additions of `configurable: true` and the setter? d3633bb

Reply:
It should still work. 👍

Reply:
Super, thanks!